### PR TITLE
Ihc updates

### PR DESCRIPTION
**Pull request by jim-bo**
@roshni-b @curlup would you mind doing a quick review of this PR.

Thanks!

### CODE REVIEW

File: cidc_schemas/prism.py [hunk 1]
```diff
@@ -356,6 +361,11 @@ def _process_field_value(
     in template schema, that allows for multi-processing of a single cell value.
     """
 
+    # skip nullable
+    if "allow_empty" in field_def:
```
Comment (curlup) [inline cidc_schemas/prism.py]: Nit: I would also check for `allow_empty`s value:
```suggestion
    if field_def.get("allow_empty"):
```

File: cidc_schemas/schemas/assays/components/imaging/ihc_entry.json [hunk 2]
```diff
@@ -5,80 +5,66 @@
...
-    "clinically_positive",
-    "percentage_viable_tissue",
-    "percentage_tumor",
-    "degree_lymphoid_infiltrate",
-    "percentage_fibrosis",
+    "cimac_id",
     "files"
-  ]
+  ],
+  "anyOf": [
```
Comment (curlup) [inline cidc_schemas/schemas/assays/components/imaging/ihc_entry.json]: nit: maybe
```suggestion
  "oneOf": [
```

File: cidc_schemas/schemas/templates/metadata/ihc_template.json [hunk 3]
```diff
@@ -7,143 +7,115 @@
...
+                "data_columns": {
+                    "Sample": {
+                        "cimac ID": {
+                            "merge_pointer": "0/cimac_id",
+                            "type_ref": "sample.json#properties/cimac_id"
+                        },
+                        "IHC image": {
+                            "merge_pointer": "0/files/ihc_image",
+                            "is_artifact": 1,
+                            "gcs_uri_format": "{protocol identifier}/{cimac id}/ihc/ihc.tiff",
```
Comment (curlup) [inline cidc_schemas/schemas/templates/metadata/ihc_template.json]: As I'm moving now the assay prefix in-front of cimac_id in all other assay templates, to be able to issue permissions with prefix like "trial/assay" 
(see here https://github.com/CIMAC-CIDC/cidc-schemas/pull/257) 
I would suggest 
```suggestion
                            "gcs_uri_format": "{protocol identifier}/ihc/{cimac id}/ihc.tiff",
```

File: cidc_schemas/schemas/templates/metadata/ihc_template.json [hunk 4]
```diff
@@ -7,143 +7,115 @@
...
+                            "type_ref": "assays/components/local_file.json#properties/file_path"
+                        }
+                    },
+                    "Chromogenic IHC": {
```
Comment (curlup) [inline cidc_schemas/schemas/templates/metadata/ihc_template.json]:
```suggestion
                    "Chromogenic IHC (please specify one score/metric)": {
```

File: cidc_schemas/schemas/assays/components/imaging/ihc_entry.json (hunk 2 above)
Comment (jim-bo) [reply cidc_schemas/schemas/assays/components/imaging/ihc_entry.json]: No they can annotate more if they want.

File: cidc_schemas/schemas/assays/ihc_assay.json [hunk 5]
```diff
@@ -1,32 +1,60 @@
...
+            ]
+        },
+        "autostainer_model": {
+            "description": "Model version of the autostainer instrument.",
+            "type": "string",
+            "enum": [
+                "Bond RX",
+                "N/A"
+            ]
+        },
```
Comment (roshni-b) [inline cidc_schemas/schemas/assays/ihc_assay.json]: `slide_scanner_model`, `staining_platform` and `autostainer_model` have been earlier defined in [image.json](https://github.com/CIMAC-CIDC/cidc-schemas/blob/master/cidc_schemas/schemas/assays/components/image.json) and I think these properties are also required for other imaging assays so it makes sense to not re-define in ihc_assay.json. You will need to update enums for `autostainer_model`.

File: cidc_schemas/schemas/assays/components/imaging/ihc_entry.json [hunk 6]
```diff
@@ -5,80 +5,66 @@
...
+      "description": "Area of PD-L1+ Inflammatory Cells over the area of TSI + IT as a percentage.",
+      "type": "number",
       "minimum": 0,
-      "exclusiveMaximum": 3
+      "maximum": 1
     },
-    "percentage_fibrosis": {
-      "description": "Percentage of fibrosis (0 - 100).",
+    "h_score": {
+      "description": "A semiquantitative approach used to assign score to tumor samples based on marker staining intensity (0, 1+, 2+, or 3+) for each cell in a fixed field.",
```
Comment (roshni-b) [inline cidc_schemas/schemas/assays/components/imaging/ihc_entry.json]: @jim-bo Can you review this description of H-score and send me an updated version? Seems like it's on a scale of 0-300.

**Pull request merged**